// next-sitemap/dist/cjs/config/index.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const _1 = require(".");
describe('next-sitemap/config', () => {
    test('defaultConfig', () => {
        expect(_1.defaultConfig).toStrictEqual({
            sourceDir: '.next',
            outDir: 'public',
            priority: 0.7,
            changefreq: 'daily',
            sitemapSize: 5000,
            autoLastmod: true,
            exclude: [],
            robotsTxtOptions: {
                policies: [
                    {
                        userAgent: '*',
                        allow: '/',
                    },
                ],
                additionalSitemaps: [],
            },
        });
    });
    test('withDefaultConfig', () => {
        const myConfig = _1.withDefaultConfig({
            sourceDir: 'custom-source',
            generateRobotsTxt: true,
            sitemapSize: 50000,
            exclude: ['1', '2'],
            robotsTxtOptions: {
                policies: [],
                additionalSitemaps: [
                    'https://example.com/awesome-sitemap.xml',
                    'https://example.com/awesome-sitemap-2.xml',
                ],
            },
        });
        expect(myConfig).toStrictEqual({
            sourceDir: 'custom-source',
            outDir: 'public',
            priority: 0.7,
            changefreq: 'daily',
            sitemapSize: 50000,
            autoLastmod: true,
            generateRobotsTxt: true,
            exclude: ['1', '2'],
            robotsTxtOptions: {
                policies: [],
                additionalSitemaps: [
                    'https://example.com/awesome-sitemap.xml',
                    'https://example.com/awesome-sitemap-2.xml',
                ],
            },
        });
    });
});
